lib/eslint.js

```diff
@@ -33,6 +33,18 @@ const assert = require("assert"),
 //------------------------------------------------------------------------------
 
 /**
+ * The result of a parsing operation from parseForESLint()
+ * @typedef {Object} CustomParseResult
+ * @property {Object} ast The ESTree AST Program node.
```

Comment:
Question: Do we have a typedef we could use for the AST? (Is it possible to pull it in from somewhere else? I don't know JSDoc very well.)